src/main/java/com/github/emcat/metric_calculators/MethodMetricsAggregator.java

```diff
@@ -0,0 +1,75 @@
+        try {
+            exception = null;
+
+            ASTMethodDeclaration methodDeclaration = getMethodDeclaration(
+                    methodDescriptor.filepath(),
+                    methodDescriptor.className(),
+                    methodDescriptor.methodName()
+            );
+
+            return metricsCalculator.calculateMethodMetrics(methodDeclaration);
+        }
+        catch (Exception exception) {
+            this.exception = new MethodProcessingException(methodDescriptor, exception);
+            return null;
+        }
+    }
+
+    public MethodProcessingException getException() {
+        return exception;
+    }
+
+    private ASTMethodDeclaration getMethodDeclaration(String filename, String className, String methodName) throws IOException {
+        ASTClassOrInterfaceDeclaration classDeclaration = getClassDeclaration(filename, className);
+        for (ASTAnyTypeBodyDeclaration classInnerDeclaration : classDeclaration.getDeclarations()) {
+            if (classInnerDeclaration.getKind() == ASTAnyTypeBodyDeclaration.DeclarationKind.METHOD) {
+                ASTMethodDeclaration methodDeclaration = (ASTMethodDeclaration) classInnerDeclaration.getDeclarationNode();
+                if (methodDeclaration.getName().equals(methodName)) {
+                    return methodDeclaration;
+                }
+            }
+        }
+
+        throw new RuntimeException(String.format("There is no method named %s in class named %s in file %s", methodName, className, filename));
+    }
+
+    private ASTClassOrInterfaceDeclaration getClassDeclaration(String filename, String className) throws IOException {
+        ASTCompilationUnit compilationUnit = getCompilationUnit(filename);
+        for (JavaNode child : compilationUnit.children()) {
+            if (child instanceof ASTTypeDeclaration) {
+                for (JavaNode grandChild : child.children()) {
+                    if (grandChild instanceof ASTClassOrInterfaceDeclaration classDeclaration) {
+                        if (!classDeclaration.isInterface() && classDeclaration.getSimpleName().equals(className)) {
+                            return classDeclaration;
+                        }
+                    }
+                }
+            }
+        }
+
+        throw new RuntimeException(String.format("There is no class named %s in file %s", className, filename));
+    }
+
+    private ASTCompilationUnit getCompilationUnit(String filename) throws IOException {
+        try (Reader sourceCodeReader = new FileReader(filename)) {
+            return (ASTCompilationUnit) javaParser.parse(filename, sourceCodeReader);
+        }
+    }
+
+    private Parser javaParser = new JavaLanguageModule().getDefaultVersion().getLanguageVersionHandler().getParser(new ParserOptions());
+    private MethodMetricsCalculator metricsCalculator = new MethodMetricsCalculator();
+    private MethodProcessingException exception = null;
```

Comment:
no need to initialize fields with null explicitly, in classes they are already initialized